src/passes/stringify-walker.h

```diff
@@ -235,10 +235,19 @@ struct HashStringifyWalker : public StringifyWalker<HashStringifyWalker> {
 struct StringifyProcessor {
   static std::vector<SuffixTree::RepeatedSubstring>
   dedupe(const std::vector<SuffixTree::RepeatedSubstring>&& substrings);
+  // Filter is the general purpose function backing subsequent filter functions.
+  // It can be used directly, but generally prefer a wrapper function
+  // to encapsulate your condition and make it available for tests
   static std::vector<SuffixTree::RepeatedSubstring>
   filter(const std::vector<SuffixTree::RepeatedSubstring>&& substrings,
          const std::vector<Expression*> exprs,
          std::function<bool(const Expression*)> condition);
+  static std::vector<SuffixTree::RepeatedSubstring>
+  filterLocalSets(const std::vector<SuffixTree::RepeatedSubstring>&& substrings,
+                  const std::vector<Expression*> exprs);
```

Comment:
Actually, if these functions don't re-use the space for the input vector, then taking them as `const vector&` should be sufficient. `vector&&` would allow you to safely reuse the input vector as the output vector, but `const vector&&` isn't really useful.

Reply:
done!